src/device/mod.rs

```diff
@@ -214,6 +214,12 @@ pub struct RequestedState {
 
     // Requested keepalive periods
     pub(crate) keepalive_periods: FeaturePersistentKeepalive,
+
+    // Use link detection for downgrades
+    pub use_link_detection_for_downgrade: bool,
+
+    // Down connections
+    pub down_connections: HashSet<PublicKey>,
```

Comment:
Maybe it would be possible to make connection management based on link status stateless? I mean why do we need a copy of link states for peers when WG already has this info?